src/org.hdfgroup.object/hdf/object/FileFormat.java

```diff
@@ -1992,6 +1992,16 @@ public void setLibBounds(String lowStr, String highStr) throws Exception {
         throw new UnsupportedOperationException("Unsupported operation. Subclasses must implement FileFormat:getLibBounds.");
     }
 
+    /**
+     * Initialize the bounds of library versions
+     *
+     * @throws Exception
+     *             The exceptions thrown vary depending on the implementing class.
+     */
+    public void initLibBounds() throws Exception {
+        throw new UnsupportedOperationException("Unsupported operation. Subclasses must implement FileFormat:initLibBounds.");
+    }
```

Comment:
I'm thinking this should only be in H5File. It's HDF5-specific and doesn't seem like it's going to be implemented by other formats. It's also probably going to always be surrounded by a cast to (H5File) before using the method anyway.

Reply:
Yes, I thought about that, but something didn't work (I think where the original call is, maybe) I will review.

Reply:
I went with the base class, because of the getLibBounds and getLibBoundsDescription functions were already in the FileFormat class. I thought it best to keep everything consistent. However, indeed initLibBounds is only called within H5File class and could be removed from the base class. getLibVerBounds is a bit different; In DefaultBaseMetaDataView it is guarded by a "isH5" but does not cast to H5File. (easy fix). In DefaultTreeView, it is not guarded and needs more investigation.

Reply:
I am thinking to just pull all trhee libver bounds functions from FileFormat class.

Reply:
However, by removing them from the base class, requires that the import H5File is then added to the hdfview classes.

Reply:
So far we have already broken the import for h5link and h5datatype objects. Maybe we should just require hdf5 and be done with it?

Reply:
Yes, including H5File is not really a problem I think because we already have H5-specific logic in a lot of places. Ideally, everything would  be hidden behind an interface and we'd never need to do any casts, but the object library/HDFView would need a little bit of architectural rework.

Reply:
I'm fine with requiring hdf5, but I think we can no longer go back. We could probably work around the current imports (although h5datatype would be very difficult now).

Reply:
BTW the major import breaks are in the TableView classes because of the data conversions.

Reply:
Don't we already require HDF5 (with HDF4 being the only optional component)? If not, then we should be able to continue to architect this so that HDF5 is only required when you use HDF5-specific components.

Reply:
Don't forget netcdf and fits support. Until this H5File issue, H5Link was needed for the metadata view and h5datatype for the data validators/conversions.

Reply:
> BTW the major import breaks are in the TableView classes because of the data conversions.

I believe I've previously considered having H4/H5 versions of the tableviews due to the file format-specific logic, but didn't try implementing that because the things that would be different were minimal enough that it seemed like kind of a pain at the time. This might be a good use case pointing towards needing that though.

Reply:
If I leave the libverbounds functions in the base class, then we would not expand the hdf5 dependency any further.

Reply:
If leaving it in the base class is the path of least pain then go ahead, but clearly there are some architecture issues with it that we should fix eventually.

Reply:
Less future pain. We may need to consider a HDF5View* package of derived classes?
The H5Datatype casts I needed to add in the data validator classes were because of references. There is probably a better way to handle those - just didn't have time.